### PR TITLE
DecisionIssue validation

### PR DESCRIPTION
**Pull request by leikkisa**
connects #9064 
connects #9172

### CODE REVIEW

File: app/models/request_issue.rb [hunk 1]
```diff
@@ -421,9 +415,6 @@ def duplicate_of_issue_in_active_review?
   end
 
   def create_decision_issues
-    # TODO: we can probably remove this error, we've learned the issue was from date formatting
-    fail NilEndProductLastActionDate, id unless end_product_establishment.result.last_action_date
-
```
Comment (leikkisa) [inline app/models/request_issue.rb]: This has been replaced with DecisionIssue validation

File: app/models/request_issue.rb [hunk 2]
```diff
@@ -449,6 +440,8 @@ def create_decision_issue_from_disposition
         participant_id: review_request.veteran.participant_id,
         disposition: contention_disposition.disposition,
         description: "#{contention_disposition.disposition}: #{description}",
+        profile_date: end_product_establishment.associated_rating&.profile_date,
+        promulgation_date: end_product_establishment.associated_rating&.promulgation_date,
```
Comment (shanear) [inline app/models/request_issue.rb]: I don't see this tested. We should probably update the test for this

File: spec/models/request_issue_spec.rb [hunk 3]
```diff
@@ -955,7 +968,8 @@
                 decision_review: review,
                 participant_id: veteran.participant_id,
                 disposition: "denied",
-                rating_issue_reference_id: contested_rating_issue_reference_id
+                rating_issue_reference_id: contested_rating_issue_reference_id,
+                end_product_last_action_date: end_product_last_action_date
```
[the remark this answers is not in the document]
Comment (leikkisa) [reply spec/models/request_issue_spec.rb]: 👍

File: app/controllers/intakes_controller.rb [hunk 4]
```diff
@@ -40,7 +40,7 @@ def review
 
   def complete
     intake.complete!(params)
-    if intake.detail.try(:processed_in_caseflow?)
+    if !intake.detail.is_a?(Appeal) && intake.detail.try(:processed_in_caseflow?)
```
[the remark this answers is not in the document]
Comment (shanear) [reply app/controllers/intakes_controller.rb]: This is good, thanks for the explainer 😄

**Comment by leikkisa** [comment]
Updated these DecisionIssues in UAT:
[9, 11, 85, 86, 87, 88, 120, 121, 122, 123]
Where the caseflow_decision_date was stored in the promulgation_date from before. I copied the decision date to caseflow_decision_date, but did not remove the promulgation_date yet.

There were no instances of this in Prod.  Also checked that these were all non-comp and that the ones that weren't included were all comp/pen.

Here is the script:

DecisionIssue.where.not(promulgation_date: nil).where(profile_date: nil).each {|di| di.update!(caseflow_decision_date: di.promulgation_date)}

**Pull request merged**